h2/src/main/org/h2/mvstore/MVStore.java

```diff
@@ -690,34 +709,35 @@ private void readStoreHeader() {
                 if (m == null) {
                     continue;
                 }
-                int blockSize = DataUtils.readHexInt(
-                        m, "blockSize", BLOCK_SIZE);
+                int blockSize = DataUtils.readHexInt(m, HDR_BLOCK_SIZE, BLOCK_SIZE);
                 if (blockSize != BLOCK_SIZE) {
                     throw DataUtils.newIllegalStateException(
                             DataUtils.ERROR_UNSUPPORTED_FORMAT,
                             "Block size {0} is currently not supported",
                             blockSize);
                 }
-                long version = DataUtils.readHexLong(m, "version", 0);
+                long version = DataUtils.readHexLong(m, HDR_VERSION, 0);
+                assumeCleanShutdown = newest == null || version == newest.version;
```

Comment:
`assumeCleanShutdown` should really be assigned before the following `if`? If so, maybe add some comment about the reason?

Reply:
where else would you assign it? After exiting the loop that flag suppose to be true if both header blocks do agree on version.

Reply:
I mean that in the `if` block below (lines 721–731) an exception is still possible due to malformed number, for example. But `assumeCleanShutdown` will be `true`. It that normal?

Reply:
You are right, I forgot about exception, thanks.